pkg/node/config_requester.go

```diff
@@ -63,37 +58,14 @@ func NewRequesterConfigWith(params RequesterConfigParams) (config RequesterConfi
 			panic(fmt.Sprintf("Failed to initialize compute config %s", err.Error()))
 		}
 	}()
-	if params.MinJobExecutionTimeout == 0 {
-		params.MinJobExecutionTimeout = DefaultRequesterConfig.MinJobExecutionTimeout
-	}
-	if params.DefaultJobExecutionTimeout == 0 {
-		params.DefaultJobExecutionTimeout = DefaultRequesterConfig.DefaultJobExecutionTimeout
-	}
-	if params.HousekeepingBackgroundTaskInterval == 0 {
-		params.HousekeepingBackgroundTaskInterval = DefaultRequesterConfig.HousekeepingBackgroundTaskInterval
-	}
-	if params.NodeRankRandomnessRange == 0 {
-		params.NodeRankRandomnessRange = DefaultRequesterConfig.NodeRankRandomnessRange
-	}
-	if params.OverAskForBidsFactor == 0 {
-		params.OverAskForBidsFactor = DefaultRequesterConfig.OverAskForBidsFactor
-	}
-	if params.MinBacalhauVersion == (model.BuildVersionInfo{}) {
-		params.MinBacalhauVersion = DefaultRequesterConfig.MinBacalhauVersion
-	}
 
-	config = RequesterConfig{
-		MinJobExecutionTimeout:             params.MinJobExecutionTimeout,
-		DefaultJobExecutionTimeout:         params.DefaultJobExecutionTimeout,
-		HousekeepingBackgroundTaskInterval: params.HousekeepingBackgroundTaskInterval,
-		JobSelectionPolicy:                 params.JobSelectionPolicy,
-		NodeRankRandomnessRange:            params.NodeRankRandomnessRange,
-		OverAskForBidsFactor:               params.OverAskForBidsFactor,
-		ExternalValidatorWebhook:           params.ExternalValidatorWebhook,
-		FailureInjectionConfig:             params.FailureInjectionConfig,
-		MinBacalhauVersion:                 params.MinBacalhauVersion,
-		RetryStrategy:                      params.RetryStrategy,
+	err = mergo.Merge(&params, getRequesterConfigParams())
```

Comment:
Does this override any unset params with the default values from `getRequesterConfigParams`?